src/tree_cmd.cpp

```diff
@@ -193,11 +196,19 @@ static void PlaceTreeGroups(uint num_groups)
 			int x = GB(r, 0, 5) - 16;
 			int y = GB(r, 8, 5) - 16;
 			uint dist = abs(x) + abs(y);
+			uint max_dist = MAX_DISTANCE_FROM_CENTER;
+
+			if (_settings_newgame.game_creation.tree_placer == TP_IMPROVED) {
+				max_dist = GB(r, 16, 3) + 8;
+			}
+
+			if (dist > max_dist) continue;
+
```

Comment:
Testing this versus "original," I cannot see a noticeable difference between the size of tree groups. You said in #12712 that you don't know why this code exists. Can you please test with and without it on Improved, to determine if you want to keep it in your PR? If you do, please show screenshots to show us its effect.

I realize that not including this limits the PR to `Change: Plant fewer solitary trees during worldgen`, but that's still worth considering, especially alongside #12712.